[Streams 🌊] Update condition editor enabling and fixes

tonyghiani opened the pull request:
## 📓 Summary

Closes #217884 

- Updates the condition editor to have a more consistent behaviour for enabled/disabled routing.
  - A more explicit tooltip is added to describe how the status flag affects the routing behaviour.
  - The status switch is visible by default, while before it was shown only in edit mode for a routing condition.
  - Fixed crashes when manually working on the syntax editor.
- Removes the routing status flag from the condition editor in the processors' config.

<img width="763" alt="Screenshot 2025-04-14 at 10 23 42" src="https://github.com/user-attachments/assets/8521739a-ac53-4751-9ad3-4400a84c5a8d" />



<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

tonyghiani committed:
refactor(streams): update condition editor

tonyghiani committed:
refactor(streams): update condition editor

tonyghiani committed:
refactor(streams): minor change

tonyghiani committed:
refactor(streams): remove issue on condition update

tonyghiani committed:
refactor(streams): update condition message

tonyghiani committed:
refactor(streams): update tootlip

elasticmachine commented:
Pinging @elastic/obs-ux-logs-team (Team:obs-ux-logs)

tonyghiani committed:
Merge branch 'main' into refactor/condition-editor-fixes

tonyghiani committed:
refactor(optimizer): update inspector limits with optimizer script

tonyghiani committed:
Merge branch 'main' into refactor/condition-editor-fixes

flash1293 commented:
Seems like something is wrong - it doesn't hide the value input anymore when a unary operator is seleted:
<img width="460" alt="Screenshot 2025-04-14 at 10 52 48" src="https://github.com/user-attachments/assets/dcf33d08-46cb-4a9c-b12f-7342b77762ff" />

Also when having invalid JSON and trying to hit save you are hit with this error (we should probably validate client side):
<img width="898" alt="Screenshot 2025-04-14 at 10 57 21" src="https://github.com/user-attachments/assets/1f190937-1259-4f4d-94ba-b9afdc8c551d" />

tonyghiani committed:
fix(streams): restore unary filter UI

tonyghiani committed:
Merge branch 'refactor/condition-editor-fixes' of github.com:tonyghiani/kibana into refactor/condition-editor-fixes

tonyghiani committed:
refactor(streams): add basic client-side validation for condition

tonyghiani committed:
refactor(streams): remove log

tonyghiani commented:
> It doesn't hide the value input anymore when a unary operator is selected

Good catch, fixed in [fix(streams): restore unary filter UI](https://github.com/elastic/kibana/pull/218055/commits/2b32fa7a2aee7367cbfb5fdc2fae298bc0362e95)

> Also when having invalid JSON and trying to hit save you are hit with this error (we should probably validate client side):

Yeah, this was something that has never been in place but deserves attention... I added a basic validation with [refactor(streams): add basic client-side validation for condition](https://github.com/elastic/kibana/pull/218055/commits/fb5de023c348b1e6454a8517f1f71ab54dfedd9d) to let the user save only valid condition, but we can surely improve this with better feedback in the long term.

You might find some issues we discussed offline have not been addressed here, those will be part of a refactor on the state management, as it relates more to that part than the individual ConditionEditor component.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292649)
* Commit: f510b8ecd433fea811d610042411296fd1fca7c1

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a1a0d3462dfd7aff84f15eeb4448d1448e73a29e) | [after](https://github.com/elastic/kibana/commit/f510b8ecd433fea811d610042411296fd1fca7c1) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 405.6KB | 405.3KB | -280.0B |



### History
* :broken_heart: [Build #292563](https://buildkite.com/elastic/kibana-pull-request/builds/292563) failed 26d7524dd78eeffb3d7a8143fb4d4af3aeeb503a

cc @tonyghiani

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292649","number":292649,"commit":"f510b8ecd433fea811d610042411296fd1fca7c1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292563","number":292563,"commit":"26d7524dd78eeffb3d7a8143fb4d4af3aeeb503a"}],"number":292649}
buildkite-pr-comment-->

flash1293 reviewed (approved):
LGTM, this is definitely improving the experience, thanks for the fixes. While testing it occurred to me that we should improve the enrichment page as well - over there disabling/enabling doesn't work but it would be cool there too and also it would be nice to disable the condition. Right now you have to manually add it back to an empty `equals` condition, which is not very intuitive. But this can happen on a separate PR, it's not making it worse over there.

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/218106">](https://github.com/elastic/kibana/pull/218106)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
